test/protocol-tests/src/test/java/software/amazon/awssdk/protocol/tests/connection/SyncClientConnectionInterruptionTest.java

```diff
@@ -0,0 +1,253 @@
+                                System.out.println(metricCollection);
+                                Optional<MetricRecord<?>> apiCallDuration =
+                                    metricCollection.stream().filter(o -> "ApiCallDuration" .equals(o.metric().name())).findAny();
+                                leaseWaitingTime.set(Duration.parse(apiCallDuration.get().value().toString()).toMillis());
+                            }
+
+                            @Override
+                            public void close() {
+                            }
+                        })
+                ));
+
+            } catch (Exception exception) {
+                exceptionInThreadRun.setException(exception);
+
+            }
+        });
+
+        leaseWaitingThread.start();
+        unInterruptedSleep(100);
+        leaseWaitingThread.interrupt();
+        leaseWaitingThread.join();
+        assertThat(leaseWaitingTime.get()).isNotEqualTo(LONG_DELAY.longValue());
+        assertThat(leaseWaitingTime.get()).isLessThan(LONG_DELAY.longValue());
+        assertThat(exceptionInThreadRun.getException()).isInstanceOf(AbortedException.class);
+        client.close();
+    }
+
+    /**
+     * Service Latency is set to high value say X.
+     * Api timeout value id set to 1/3 of X.
+     * And we interrupt the thread at 90% of X.
+     * In this case since the ApiTimeOut first happened we should get ApiTimeOut Exception and not the interrupt.
+     */
+    @Test
+    void interruptionDueToApiTimeOut_followed_byInterruptCausesOnlyTimeOutException() throws InterruptedException {
+        SdkHttpClient httpClient = ApacheHttpClient.create();
+        Integer SERVER_RESPONSE_DELAY = 3000;
+        stubPostRequest("/2016-03-11/allTypes", aResponse().withFixedDelay(SERVER_RESPONSE_DELAY), SAMPLE_BODY);
+        ExceptionInThreadRun exception = new ExceptionInThreadRun();
+        ProtocolRestJsonClient client =
+            getClient(httpClient, Duration.ofMillis(10)).overrideConfiguration(o -> o.retryPolicy(RetryPolicy.none())).build();
+        unInterruptedSleep(100);
+        // We need to creat a separate thread to interrupt it externally.
+        Thread leaseWaitingThread = new Thread(() -> {
+            try {
+                client.allTypes(l -> l.overrideConfiguration(b -> b.apiCallAttemptTimeout(Duration.ofMillis(SERVER_RESPONSE_DELAY / 3))));
+            } catch (Exception e) {
+                exception.setException(e);
+            }
+        });
+        leaseWaitingThread.start();
+        unInterruptedSleep(SERVER_RESPONSE_DELAY - SERVER_RESPONSE_DELAY / 10);
+        leaseWaitingThread.interrupt();
+        leaseWaitingThread.join();
+        assertThat(exception.getException()).isInstanceOf(ApiCallAttemptTimeoutException.class);
+        client.close();
+    }
+
+
+    @Test
+    void sdkClientInterrupted_while_connectionIsInProgress() throws InterruptedException {
```

Comment:
How is this different from `connectionPoolsGetsReusedWhenInterruptedWith_1_MaxConnection`?

Reply:
This was added to test default settings with UrlConnection , its no longer needed.